_posts/2018-06-15-cocoapods-keys-react-native.markdown

```diff
@@ -0,0 +1,156 @@
+
+[/Example/Podfile](https://github.com/artsy/emission/blob/4a2a3e9260e97d791536cf38376a06b0ad0946a8/Example/Podfile#L63):
+
+```diff
+plugin 'cocoapods-keys', {
+  :target => 'Emission',
+  :keys => [
+    'ArtsyAPIClientSecret',   # Authing to the Artsy API
+    'ArtsyAPIClientKey',      #
++    'StripePublishableKey',
+  ]
+}
+```
+
+---
+
+#### 2. Configure the library to consume our new key
+
+We'll need to update the `initWithUserId...` function -- one **fun** part of adjusting to Objective-C is that rather than named funtions, we just refer to them by their entire signatures -- to expose the new key as a property and add it to `constantsToExport`.
+
+Note that this is happening in our _Emission Pod_; The pod now expects that key to be available in our _consuming_ Example app as defined above.
+
+[/Pod/Classes/Core/AREmission.h](https://github.com/artsy/emission/blob/4a2a3e9260e97d791536cf38376a06b0ad0946a8/Pod/Classes/Core/AREmission.h#L17-L34):
+
+```diff
+// ENV Variables
++ @property (nonatomic, copy, readonly, nullable) NSString *stripePublishableKey;
+
+# ...
+
+ - (instancetype)initWithUserID:(NSString *)userID
+           authenticationToken:(NSString *)token
+                     sentryDSN:(nullable NSString *)sentryDSN
++         stripePublishableKey:(NSString *)stripePublishableKey
+              googleMapsAPIKey:(nullable NSString *)googleAPIKey
+                    gravityURL:(NSString *)gravity
+                metaphysicsURL:(NSString *)metaphysics
+                     userAgent:(NSString *)userAgent;
+```
+
+`AREmission`'s implementation (.m) needs to be configured to take this new key- It will be exported to our our React Native components as `Emission`. We make our intializer match the signature defined in the header (.h) file, and add an instance `_stripePublishableKey` to match the `@property` declaration.
+[/Pod/Classes/Core/AREmission.m](https://github.com/artsy/emission/blob/4a2a3e9260e97d791536cf38376a06b0ad0946a8/Pod/Classes/Core/AREmission.m#L24-L60):
+
+```diff
+ @implementation AREmissionConfiguration
+ RCT_EXPORT_MODULE(Emission);
+ # ...
+
+ - (NSDictionary *)constantsToExport
+ {
+   return @{
++    @"stripePublishableKey": self.stripePublishableKey ?: @"",
+     # ...lots more
+   };
+ }
+
+ - (instancetype)initWithUserID:(NSString *)userID
+            authenticationToken:(NSString *)token
+                      sentryDSN:(NSString *)sentryDSN
++          stripePublishableKey:(NSString *)stripePublishableKey
+               googleMapsAPIKey:(NSString *)googleAPIKey
+                     gravityURL:(NSString *)gravity
+                 metaphysicsURL:(NSString *)metaphysics
+                      userAgent:(nonnull NSString *)userAgent
+ {
+     self = [super init];
+     _userID = userID.copy;
++    _stripePublishableKey = stripePublishableKey.copy;
```

Comment:
Do we really call `copy` as a property? (with dot syntax) Because that’s not really idiomatic objc, dot syntax should really only be used for ‘properties’ of an object, not to invoke methods.

Reply:
Welllll, I mean the blog post has the code that we ship with Emission. I've never really worried about the distinction between the two, but I'm strictly an Objective-C 2.0+ developer. We can change it here, but if we do we should change it in Emission as well.

Reply:
Oh no, sorry, I meant we should change it in Emission, it’s just surfacing to me now through this blog post :) I think we should fix it in both locations, we don’t want objc devs to hone in on irrelevant stuff.